src/bitmap/inherent.rs

```diff
@@ -64,17 +58,31 @@ impl RoaringBitmap {
     /// assert!(rb.contains(3));
     /// assert!(!rb.contains(4));
     /// ```
-    pub fn insert_range(&mut self, range: Range<u64>) -> u64 {
-        assert!(
-            range.end <= u64::from(u32::max_value()) + 1,
-            "can't index past 2**32"
-        );
-        if range.is_empty() {
+    pub fn insert_range<R: RangeBounds<u32>>(&mut self, range: R) -> u64 {
+        let start = match range.start_bound() {
+            Bound::Included(value) => *value,
+            Bound::Excluded(value) => match value.checked_add(1) {
+                Some(value) => value,
+                None => return 0,
+            },
+            Bound::Unbounded => 0,
+        };
+
+        let end = match range.end_bound() {
+            Bound::Included(value) => match value.checked_add(1) {
+                Some(value) => value,
+                None => return 0,
+            },
```

Comment:
I find it a bit strange that of these three hypothetical test cases, the first two pass and the third does not:
```rust
#[test]
fn test_insert_max_u32() {
    let mut b = RoaringBitmap::new();
    let inserted = b.insert(u32::MAX);
     // We are allowed to add u32::MAX
    assert!(inserted);
}

#[test]
fn test_insert_range_zero_inclusive() {
    let mut b = RoaringBitmap::new();
    let inserted = b.insert_range(0..=0);
    // `insert_range(value..=value)` appears equivalent to `insert(value)`
    assert_eq!(inserted, 1);
    assert!(b.contains(0));
}

#[test]
fn test_insert_range_max_u32_inclusive() {
    let mut b = RoaringBitmap::new();
    let inserted = b.insert_range(u32::MAX..=u32::MAX);
     // But not equivalent for u32::MAX
    assert_eq!(inserted, 1); // Fails - left: 0, right: 1
    assert!(b.contains(u32::MAX));
}
```

Reply:
To add onto this, since it does not seem possible with `insert_range` to add `u32::MAX` (any `Range` argument with an `end_bound` of `Inclusive(u32::MAX)` will insert 0 elements), if this is intended behavior, wouldn't this function be able to return a `u32`?

Reply:
Hello 👋 @tomjw64

Hum... Thank you very much, I borrowed your tests and will investigate this bug.
Indeed, this is not correct, we should be able to insert up to `u32::MAX` by using an inclusive bound.